### ISSUE TITLE
Clicking while an app is launching causes multiple windows to open

### ISSUE THREAD

**Issue by nkkollaw** [elementary/gala#130]
This is very annoying to me.

Apps take a while to open here, and sometimes I click on the icon in Plank multiple times.

This causes multiple windows to be open, and I absolutely never mean to do that.

I believe the icon/window/app should be in a "loading" state, and multiple clicks to the Plank icon should be ignored if the app isn't open.

I'm not sure if this is a Gala or Plank issue, though.

**Comment by Philip-Scott** [comment]
I don't think this is something that the window manager has to take care off. Apps should just open fast :) If you're using an app that's opening slow, file a report to them

**Comment by nkkollaw** [comment]
LOL

**Comment by artemanufrij** [comment]
@nkkollaw have you this behavior on all apps? almost all apps open instandly on my system (exclude epiphany and noise)

**Comment by donadigo** [comment]
I have to disagree on the stance that apps should open fast. What does _fast_ mean? 1 second, 2 seconds? To solve this issue by making apps _fast_ they would need to open instantly without any interval. If apps open fast on your system, that's good, but it's only your system. Even if you would make the app super optimized it still has to be read and loaded into memory. A system installed on a SSD will load apps relatively quickly while normal hard drives and some slower ones can take their time to load all the resources the app needs, in some cases like opening a browser this can take much longer than instant. On my system e.g: Gtk apps usually take 3 - 5 seconds with a normal HDD.

**Comment by artemanufrij** [comment]
@donadigo I think @nkkollaw means that we need an _indicator_ like "app is starting". Sometimes it's not clear (if starting takes > 3 seconds) is the app starting or not. Noise takes >10 seconds for starting on my system (SSD/i7/16GB) and you have no feedback is it starting/crashed/etc. or not...

**Comment by peteruithoven** [comment]
Since the title actually is about "causes multiple windows to open' I'm curious with what app that happens, because for me, when the app is already open it hides when clicking it's icon again.

**Comment by donadigo** [comment]
@artemanufrij yes I know about that.

@peteruithoven: the problem is not with an app that's already open, but one that's still opening and haven't shown any window yet.

**Comment by nkkollaw** [comment]
I've moved to Xubuntu because since I've opened this issue and filed a few issues similar to this one I found that elementary devs have absolutely no idea what they're doing. 

They take away features people need and want (minimize? desktop? volume control in video player, etc.?), and dismiss issues like this by saying crap like "apps should open fast, we're not doing anything about the fact that if I click on a launcher 100 times while the app isn't loaded it opens 100 instances of the app".

I was running away from the "new" Apple that removes headphone jacks and ports and ships $2000 laptops with ridiculously low battery life just to make them thinner, and makes software unusable by hiding controls because they're not "pretty", just to find another bunch of people that think the same way, but with no resources and therefore half-baked software with 3-years release cycles.

Done with the rant, but WTF, guys! Get your shit together!
